app/main.py

```diff
@@ -1,3 +1,76 @@
+from __future__ import annotations
+
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int or float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, int, float]) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Union[Distance, int, float]) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km += other
+            return self
+
+        self.km += other.km
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km *= other
+            return self
+
+        raise TypeError(f"{other} must be number of type: int or float")
+
+    def __truediv__(self, other: int | float) -> Distance:
+        if other == 0:
+            raise TypeError(f"{other} cannot be division on zero!")
```

Comment:
Instead of raising a `TypeError`, a `ZeroDivisionError` should be raised when dividing by zero, as this is the standard exception for such cases in Python.